test/unit/merge.js

```diff
@@ -18,6 +18,46 @@ const pouchHelpers = require('../support/helpers/pouch')
 const dbBuilders = require('../support/builders/db')
 const MetadataBuilders = require('../support/builders/metadata')
 
+/** Returns an object describing the side-effects of a Merge.
+ *
+ * The returned object has the following properties:
+ *
+ * - `savedDocs`: Which docs were passed to `Pouch#put()`
+ * - `resolvedConflicts`: Which conflits were resolved on which side
+ *
+ * The given `Merge` instance is expected to have its `#resolveConflictAsync()`
+ * method wrapped with `sinon.spy()`.
+ * The given `Pouch` instance is expected to have its `#put()` method spied in
+ * the same way.
+ *
+ * FIXME: `Pouch#bulkDocs()` are not yet included in `mergeSideEffects()`.
+ * FIXME: `Pouch#remove()` is not yet included in `mergeSideEffects()`.
+ */
+function mergeSideEffects ({merge, pouch} /*: * */) {
+  return {
+    savedDocs: pouch.put.args.map(_.first),
+    resolvedConflicts: merge.resolveConflictAsync.args.map(([side, doc, existing]) =>
```

Comment:
Spying Pouch calls may not be the more reliable way to identify side-effects.
But I believe it should be easy to improve later when we have a better idea (spy ids to Pouch#get the relevant docs? Or use Pouch#db.changes()?).